src/main/scala/vsys/settings/WebhookEventRules.scala

```diff
@@ -94,57 +128,79 @@ object ExcludeTypes extends RuleConfigReader {
   }
 }
 
-case class AmtGT(value: Long) extends WebhookEventRules
+case class Amount(value: Seq[WebhookEventRules]) extends WebhookEventRules {
+  override def applyRule(height: Long, tx: ProcessedTransaction, accs: Set[Address]): Boolean = {
+    val (amt, fee) = TransactionHelper.extractAmtFee(tx)
+    val feeRule = value.filter(_.isInstanceOf[AmtWithFee])
 
-object AmtGT extends RuleConfigReader {
-  override val field = "amount.gt"
-
-  override def fromConfig(config: Config): Option[AmtGT] = {
-    read[Long](config).map(AmtGT(_))
+    if (feeRule.nonEmpty && feeRule(0).applyRule(height, tx, accs)) {
+      // withFee
+      value.filterNot(_.isInstanceOf[AmtWithFee]).foldLeft(true)((accum, rule) =>
+        rule.applyRule(amt+fee, tx, accs) && accum)
+    } else {
+      // not withFee
+      value.filterNot(_.isInstanceOf[AmtWithFee]).foldLeft(true)((accum, rule) =>
+        rule.applyRule(amt, tx, accs) && accum)
+    }
   }
 }
 
-case class AmtLT(value: Long) extends WebhookEventRules
+object Amount extends RuleConfigReader {
```

Comment:
Add `applyRule()` in Amount since AmtWithFee is not a rule.

Reply:
np